src/libsync/propagateuploadng.cpp

```diff
@@ -502,4 +502,16 @@ void PropagateUploadFileNG::slotUploadProgress(qint64 sent, qint64 total)
     }
     propagator()->reportProgress(*_item, _sent + sent - total);
 }
+
+
+void PropagateUploadFileNG::abort()
+{
+    foreach (AbstractNetworkJob *job, _jobs) {
+        // Abort only PUT and MKDIR jobs, since abording MoveJob
+        // might result in conflict.
+        if (job->reply() && !job->inherits("OCC::MoveJob")) {
```

Comment:
nitpick: use qobject_cast   (more efficient and also compile-time checked)

Also, we shold probzbly decrease the timeout.  For example, if the abort happens because the connection  was detected to be dropped, we should not wait a full HTTP timeout again, IMHO.